pkg/cloudprovider/external_test.go

```diff
@@ -193,8 +246,8 @@ func TestIsCloudProviderExternal(t *testing.T) {
 		},
 		expected: false,
 	}, {
-		name:     "FeatureSet: CustomNoUpgrade (With External Feature Gate), Platform: None",
-		platform: configv1.NonePlatformType,
+		name:   "Platform status is empty",
```

Comment:
I think in this case we expect an error right? This is probably why the unit tests are failing?

Reply:
No, the tests are failing because of the unmerged dependency https://github.com/openshift/api/pull/860 And if the status if empty then it is simply not an AzureStackHub (yet or ever) so we just return false.

Reply:
If the status is empty, we don't know what the platform is at all, you added an `if status == nil { return err}` at the top of the function, we enter that block in this test case, this definitely needs to be checking for an error, but it isn't right now 🤔